lib/fuse_lowlevel.c

```diff
@@ -3353,6 +3353,7 @@ int fuse_req_getgroups(fuse_req_t req, int size, gid_t list[])
 		goto retry;
 	}
 
+	buf[ret] = '\0';
```

Comment:
Thanks, we need to add in support for FUSE_CREATE_SUPP_GROUP to be able to deprecate this function.

Reply:
That's not quite enough to deprecate it, since FUSE_CREATE_SUPP_GROUP only sends a single group currently, and only in case of creates.

But now that the API infrastructure is there it would not be too hard to send all groups in all cases.  The question is whether that's useful for anybody or not.

Reply:
I also see
```
/*
 * Code shared between mknod, mkdir, symlink and link
 */
static int create_new_entry(struct fuse_mount *fm, struct fuse_args *args,
                            struct inode *dir, struct dentry *entry,
                            umode_t mode)
```

But yeah, plain open is missing. For our file systems we typically fetch groups on the server side to avoid large rpcs - I guess we could add it for open and others when someone needs it.
What we could do on adding FUSE_CREATE_SUPP_GROUP to libfuse, is to get the groups from the request when it has it and only otherwise from /proc.